pgmpy/estimators/ExpertKnowledge.py

```diff
@@ -63,6 +79,81 @@ def _validate_edges(self, edge_list):
         else:
             return edge_list
 
+    def _validate_temporal_order(self, nodes):
+        """
+        Method to check consistency of temporal order with nodes of a graph.
+
+        The temporal order, if specified by the user, is currently used by the PC
+        algorithm. The temporal order of all nodes present in the graph/dataset
+        need to be present in the ExpertKnowledge instance.
+
+        Parameters
+        ----------
+        nodes: iterable
+            A collection of nodes present in a dataset/graph object.
+        """
+        if self.temporal_order == [[]]:
+            return
+
+        tier_set = set()
+        for tier in self.temporal_order:
+            tier_set = tier_set.union(tier)
+
+        if tier_set != set(nodes):
+            raise ValueError(
+                f"Missing nodes in temporal order - {set(nodes) - tier_set}"
+            )
+
+    def _get_temporal_ordering(self, temporal_order):
+        """
+        Method to check consistency of temporal order with nodes of a graph.
+
+        The temporal order, if specified by the user, is currently used by the PC
+        algorithm. The temporal order of all nodes present in the graph/dataset
+        need to be present in the ExpertKnowledge instance.
+
+        Parameters
+        ----------
+        temporal_order: list of lists
+            The temporal ordering of variables according to prior knowledge.
+
+        Returns
+        --------
+        temporal_ordering: dict
+            Dictionary with the tier (0, 1, 2, 3 etc.) for each node.
+        """
+        if type(temporal_order) != list:

```

Comment:
Other types should also be allowed. A better way is to test for `hasattr(temporal_order, "__iter__")`